backend/src/gamestate/game_repo.rs

```diff
@@ -0,0 +1,46 @@
+use crate::gamestate::game::Game;
+use actix_web::error::ErrorBadRequest;
+use anyhow::anyhow;
+use async_trait::async_trait;
+use jwt_simple::prelude::*;
+use nanoid::nanoid;
+use std::fmt::Error;
+use std::sync::Arc;
+
+#[async_trait]
+pub trait GameRepo {
+    async fn create_game(&mut self, name: String) -> anyhow::Result<(Game)>;
+}
+
+#[derive(Clone)]
+pub struct PostgresGameRepo {
+    games: Vec<Game>,
+}
+
+impl PostgresGameRepo {
+    pub fn new(games: Vec<Game>) -> Self {
+        Self { games }
+    }
+}
+
+#[async_trait]
+impl GameRepo for PostgresGameRepo {
+    async fn create_game(&mut self, name: String) -> anyhow::Result<(Game)> {
+        if !name.is_empty() {
```

Comment:
Would flip this condition to achieve an early return.

Reply:
Souhlas, navíc ověřování jména by mělo být na nějaké vyšší vrstvě (v handleru, případně by mohla být v herní logice), určitě ne v databázi.